config/index.js

```diff
@@ -25,20 +33,37 @@ function printConfig (config) {
 *    ]
 *
 */
-module.exports = function (config, overrides) {
-  config.load = function () {
+module.exports = function (config, configDirPath, overrides) {
+  if (!overrides && Array.isArray(configDirPath)) {
+    overrides = configDirPath
+    configDirPath = null
+  }
+
+  assert(!overrides || Array.isArray(overrides),
+    'if present, overrides must be an array')
+
+  config.load = async function () {
     if (config.applyDefaults) {
       config.applyDefaults()
     }
 
+    let allOverrides = []
+    if (configDirPath) {
+      const serviceEnv = BaseServiceConfig.getServiceEnvironment()
+      const env = (serviceEnv && serviceEnv.toLowerCase()) || 'localhost'
```

Comment:
Chicken and egg problem... can't read from config environment value since it is always the default value at this point - there could be an override to come which hasn't been applied yet.

Alternatively, I could move this and only apply values which are not in `overrides`.